### PR TITLE
[docs](clion) Add Clion remote development and debugging documents.

### PR DESCRIPTION
**Pull request by kaka11chen**
# Proposed changes

Add Clion remote development and debugging documents.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: docs/zh-CN/community/developer-guide/be-clion-dev.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+## 本地 Clion 安装配置远程开发环境
+
+1. 在本地下载安装 Clion，导入 Doris BE 代码。
+
+2. 在本地设置远程开发环境。
+在 Clion 中打开 **Preferences -> Build, Execution, Deployment -> Deployment** 中添加远程开发环境。
+使用 **SFTP** 来添加一个远程开发服务器的相关连接登陆信息。设置 **Mappings** 路径。
+比如 Local Path 为本地路径 `/User/kaka/Programs/doris/be`，Deployment Path 为远程服务器路径 `/mnt/datadisk0/chenqi/clion/doris/be`。
```
Comment (morningman) [inline docs/zh-CN/community/developer-guide/be-clion-dev.md:48]: 图和文字直接插入一个空行。

图和图之间也插入一个空行

File: docs/zh-CN/community/developer-guide/be-clion-dev.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+![Deployment1](/images/clion-deployment1.png)
+![Deployment2](/images/clion-deployment2.png)
+
+3. 将远程服务器上编译完成的 `gensrc` 路径，比如 `/mnt/datadisk0/chenqi/doris/gensrc` 拷贝到 **Deployment Path** 的上一级目录。
+比如拷贝完最终的目录为远程服务器路径 `/mnt/datadisk0/chenqi/clion/doris/gensrc`。
+```
+cp -R /mnt/datadisk0/chenqi/doris/gensrc /mnt/datadisk0/chenqi/clion/doris/gensrc
+```
+
+4. 在 Clion 中打开 **Preferences -> Build, Execution, Deployment -> Toolchains** 中添加远程环境的相关 Toolchains，比如 cmake、gcc、g++、gdb 等。
```
Comment (morningman) [inline docs/zh-CN/community/developer-guide/be-clion-dev.md:58]: 标题和正文直接插入一个空行

File: docs/zh-CN/community/developer-guide/be-clion-dev.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+**其中最关键的一点是需要在** Environment file **中 填写远程服务器 Doris 代码中的** env.sh **文件路径。**
+![Toolchains](/images/clion-toolchains.png)
+
+5. 在 Clion 中修改 `be/CmakeLists.txt`，修改 `$ENV{JAVA_HOME}` 为远程服务器上的 `JAVA_HOME` 路径。 (否则会找不到 jni.h)
+
+6. 在 Clion 中右键点击 **Load Cmake Project**。此操作会同步代码到远程服务器上，并且调用生成相关 Cmake Build Files。
+
+## 本地 Clion 运行调试远程 BE
+
+1. 在 **Preferences -> Build, Execution, Deployment -> CMake** 中配置 CMake。可以配置类似于 Debug / Release 等不同的 Target， 其中 **ToolChain** 需要选择刚才配置的。
```
Comment (morningman) [inline docs/zh-CN/community/developer-guide/be-clion-dev.md:68]: 空行

**Review by morningman — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Pull request merged**